### PR TITLE
Improve the Return Type Validation in GraphQL Interceptors

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#4255](https://github.com/ballerina-platform/ballerina-standard-library/issues/4255)
Fixes: [#4490](https://github.com/ballerina-platform/ballerina-standard-library/issues/4490)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

### CODE REVIEW

File: ballerina/response_formatter.bal [hunk 1]
```diff
@@ -64,7 +64,7 @@ class ResponseFormatter {
                 } else if selection is parser:FieldNode {
                     __Type fieldType = self.getFieldType(selection.getName(), parentType, onType);
                     anydata|anydata[] fieldResult = ();
-                    if data.hasKey(selection.getAlias()) {
+                    if data.hasKey(selection.getAlias()) || data.hasKey(selection.getName()) {
```
Comment (ThisaruGuruge) [inline ballerina/response_formatter.bal:67]: Isn't this lead to a bug? What if a certain field is required twice, once with an alias and once without an alias? Do we have a test for that?

File: ballerina/response_formatter.bal [hunk 2]
```diff
@@ -178,6 +184,15 @@ class ResponseFormatter {
             return <__Field>getFieldFromFieldArray(fields, fieldName);
         }
     }
+
+    isolated function getFieldValue(Data data, parser:FieldNode fieldNode) returns anydata|anydata[] {
+        if data.hasKey(fieldNode.getAlias()) {
+            return data[fieldNode.getAlias()];
+        } else if data.hasKey(fieldNode.getName()) {
```
Comment (ThisaruGuruge) [inline ballerina/response_formatter.bal:191]: Do we need the `else if` block here? If `data.hasKey(fieldNode.getAlias())` fails, it must have `fieldNode.getName()` key right?

File: ballerina/response_formatter.bal (hunk 1 above)
Comment (DimuthuMadushan) [reply ballerina/response_formatter.bal:67]: Do you mean a query like this? https://github.com/ballerina-platform/module-ballerina-graphql/pull/1440/files#diff-0219b63c80c6cac590137488c0e68e6491fe15e99f0305e41dd5643090116d17

Comment (ThisaruGuruge) [reply ballerina/response_formatter.bal:67]: That's exactly what I was looking for!

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Pull request merged**